java/src/jmri/jmrit/display/layoutEditor/LayoutBlock.java

```diff
@@ -2580,13 +2596,17 @@ private void addThroughPath(Block srcBlock, Block dstBlock, LayoutEditor panel)
                 log.info("sizes are not the same therefore, we will do some further checks");
             }
             List<LayoutTurnout> maxt;
-            List<Integer> maxtSet;
+            List<Integer> maxtSet = new ArrayList<>();
```

Comment:
This is *not* the correct fix for this problem and should not be done. This is not correct because manually copying complete lists should never be done, especially since all copies come from the same, no reason given, recently changed to be immutable list that should be mutable. Please fix this at the source, not at the side effects.

Reply:
While I agree this isn't the fix I'll disagree that the fix is to make the original List mutable and modify it. According to the comment at line 2565: "We know that a path that contains a double cross-over, is not reported correctly, therefore we shall do some additional checks and add it.". And line 2590: "The double crossover gets reported in the opposite setting.". Therefore lines 2591-3 swap the CLOSED (2) & THROWN (4) values for maxtSet(i). Modifying the original List may have unknown side effects (for example if this same list is passed to us again are we going to swap the CLOSED/THROWN values again?). I believe the correct behavior/fix is to swap the CLOSED/THROWN values in set (based on maxtSet.get(i)) and pass that to addThroughPathPostChecks instead of maxt. I believe this was the original authors intent (otherwise why is the set variable there?)… they just forgot to finish what they started.

Reply:
Unless you (@rhwood) are saying that the original list should be "fixed" to report double crossovers in the correct setting?

Reply:
(In which case it wouldn't need to be modified here.)

Reply:
It needs to copy the list because only sometimes (double crossover) it needs to change the list before passing it on to other code.

Reply:
Accepted the changes from Randal. I also tested his update to master against the test files and they work. So this PR will only hold javadoc changes.